### PR TITLE
[Dashboard]Don't set node actors when node_id of actor is Nil

### PR DESCRIPTION
**Pull request by WangTaoTheTonic**
## Why are these changes needed?
When actor is not ALIVE, normally it's raylet id is not set when published. The raylet id would be Nil("ffffffffffffffffffffffffffffffffffffffffffffffffffffffff", aka "f" * 56).

We should not update node_actors when the raylet id is Nil.

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by rkooo567** [comment]
Doesn't this mean we cannot get dead actors information?

**Comment by WangTaoTheTonic** [comment]
> Doesn't this mean we cannot get dead actors information?

We can still get dead actors, no matter before or after this change, as event an actor is dead, its raylet id still is valid(not empty).

This pr fix bug that when a new actor coming, it carries empty raylet id because it has not been allocated yet. Then the dashboard will put information of this actor into `node_actors` with an empty node id, which means all new actors would be put into that map unnecessarily, causing memory leak.

File: dashboard/modules/stats_collector/stats_collector_head.py [hunk 1]
```diff
@@ -233,9 +233,11 @@ def _process_actor_table_data(data):
                 # Update actors.
                 DataSource.actors[actor_id] = actor_table_data
                 # Update node actors.
-                node_actors = dict(DataSource.node_actors.get(node_id, {}))
-                node_actors[actor_id] = actor_table_data
-                DataSource.node_actors[node_id] = node_actors
+                if node_id != "f" * 56:
```
Comment (fyrestone) [inline dashboard/modules/stats_collector/stats_collector_head.py:236]: It's better to set a const to `stats_collector_consts.py`
```python
NIL_NODE_ID = ray.NodeID.nil().hex()
```
then check `if node_id != stats_collector_consts.NIL_NODE_ID`.

Comment (WangTaoTheTonic) [reply dashboard/modules/stats_collector/stats_collector_head.py:236]: Sounds like a good idea!

**Comment by fyrestone** [comment]
> > Doesn't this mean we cannot get dead actors information?
> 
> We can still get dead actors, no matter before or after this change, as even an actor is dead, its raylet id still is valid(not empty).
> 
> This pr fix bug that when a new actor coming, it carries empty raylet id because it has not been allocated yet. Then the dashboard will put information of this actor into `node_actors` with an empty node id key, which means all new actors would be put into that map unnecessarily, causing memory leak.

Good catch! The value of `DataSource.node_actors[NIL_NODE_ID]` will be leaking in such case. Actually,  none of the actor info will be deleted from Redis, so this is a leaky system. But we can avoid the leak of `DataSource.node_actors[NIL_NODE_ID]`.

File: dashboard/modules/stats_collector/stats_collector_head.py (hunk 1 above)
Comment (fyrestone) [reply dashboard/modules/stats_collector/stats_collector_head.py:236]: Can you add this check to `GetAllActorInfo` processing logic, too? Then, the assertion could be added to the `test_stats_collector.py`:
```python
response = requests.get(webui_url + "/test/dump?key=node_actors")
response.raise_for_status()
result = response.json()
assert stats_collector_consts.NIL_NODE_ID not in result["data"]["node_actors"]
```

**Comment by rkooo567** [comment]
Ping me after addressing @fyrestone's comment!

**Comment by rkooo567** [comment]
I will merge upon his approval

**Review by fyrestone — APPROVED** [review]
LGTM

**Comment by WangTaoTheTonic** [comment]
@rkooo567 It's ok to go!

**Pull request merged**